Objects/unicodeobject.c

```diff
@@ -2217,6 +2226,58 @@ PyUnicode_FromWideChar(const wchar_t *u, Py_ssize_t size)
     if (size == 0)
         _Py_RETURN_UNICODE_EMPTY();
 
+#if defined(__sun) && defined(__SVR4)
+    /* Check whether current locale uses UTF to encode symbols */
+    if (!codeset_is_utf8_compatible()) {
+
```

Comment:
Can you please move this code path into a separated function? (the if body)

I suggest to add an internal function to Python/fileutils.c to convert a wchar_t* string to a char32_t* string. The caller would be responsible to release the create string. It would raise an exception and return NULL on error.

Reply:
Done. Hopefully, I named it correctly and placed it in the correct spot.